zug/util.hpp

```diff
@@ -112,6 +114,38 @@ struct composed
     }
 };
 
+template <typename T>
+constexpr bool is_composed_v = false;
+
+template <typename... Fns>
+constexpr bool is_composed_v<composed<Fns...>> = true;
+
+template <template <typename...> class Composed, typename... Fns>
```

Comment:
I think you do not need the template parameter, you can just use `typename ComposedFns` as argument since you do not need to know what the individual `Fns` are. Also I am not sure the current instantiation is really doing perfect forwarding or move-references only.

Reply:
Also, if you use SFINAE using a default template argument instead of the return you can avoid the `decltype()` thing. I mean, concretely:
```
    template <typename ComposedFns, 
                    std::enable_if_t<is_composed_v<ComposedFns>, int>=0>
    delctype(auto) to_function_tuple(...) { return ...; }
```

Reply:
> I think you do not need the template parameter, you can just use `typename ComposedFns` as argument since you do not need to know what the individual `Fns` are.

Yup - this was a remnant of a previous iteration. Nice catch!